File: tests/fixtures/python/basic-crypto/cipher/aes.py
"""AES encryption functions."""

from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes  # pyright: ignore[reportMissingImports]
from cryptography.hazmat.backends import default_backend  # pyright: ignore[reportMissingImports]

KEY_SIZE_128 = 16
KEY_SIZE_256 = 32


def create_aes_128_cipher(key: bytes, iv: bytes):
    """Create an AES-128 cipher in CBC mode."""
    cipher = Cipher(algorithms.AES(key[:16]), modes.CBC(iv), backend=default_backend())
    return cipher.encryptor()


def create_aes_256_cipher(key: bytes, iv: bytes):
    """Create an AES-256 cipher with literal key size."""
    cipher = Cipher(algorithms.AES(key[:32]), modes.CBC(iv), backend=default_backend())
    return cipher.encryptor()


def create_aes_gcm(key: bytes, nonce: bytes):
    """Create an AES-GCM cipher."""
    cipher = Cipher(algorithms.AES(key), modes.GCM(nonce), backend=default_backend())
    return cipher.encryptor()

File: tests/fixtures/go/basic-crypto/pkg/cipher/aes.go
package cipher

import (
	"crypto/aes"
	"crypto/cipher"
)

const KeySize128 = 16
const KeySize256 = 32

// NewAES128 creates an AES-128 cipher
func NewAES128(key []byte) (cipher.Block, error) {
	if len(key) < KeySize128 {
		return nil, nil
	}
	return aes.NewCipher(key[:KeySize128])
}

// NewAES256 creates an AES-256 cipher with literal key size
func NewAES256(key []byte) (cipher.Block, error) {
	if len(key) < 32 {
		return nil, nil
	}
	return aes.NewCipher(key[:32])
}

// CreateGCM creates a GCM cipher from a block
func CreateGCM(block cipher.Block) (cipher.AEAD, error) {
	return cipher.NewGCM(block)
}

